core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/registry/DefaultCodecRegistry.java

```diff
@@ -48,7 +48,24 @@
   private final LoadingCache<CacheKey, TypeCodec<?>> cache;
 
   /**
-   * Creates a new instance, with some amount of control over the cache behavior.
+   * Creates a new instance that accepts user codecs, with the default built-in codecs and the
+   * default cache behavior.
+   */
+  public DefaultCodecRegistry(@NonNull String logPrefix) {
+    this(logPrefix, CodecRegistryConstants.PRIMITIVE_CODECS);
+  }
+
+  /**
+   * Creates a new instance that accepts user codecs, with the given built-in codecs and the default
+   * cache behavior.
+   */
+  public DefaultCodecRegistry(@NonNull String logPrefix, @NonNull TypeCodec<?>[] primitiveCodecs) {
+    this(logPrefix, 0, null, 0, null, primitiveCodecs);
+  }
```

Comment:
Unlike `CachingCodecRegistry`, I didn't preserve backward compatibility for this class's constructors. It's much less likely that someone extended it.

One catch is that the last argument of `DefaultCodecRegistry(String, TypeCodec<?>[])` now corresponds to primitive codecs (vs. user codecs in 4.2.0). But if someone runs into that things will blow out pretty fast, there's no risk that it goes undetected.